server/services/offender/responses/approvedVisitors.js

```diff
@@ -1,15 +1,10 @@
-const { capitalizePersonName } = require('../../../utils/index');
+const { capitalizePersonName, sortBy } = require('../../../utils/index');
 
-const sortContacts = ({ createDateTime: a }, { createDateTime: b }) =>
-  new Date(b) - new Date(a);
-
-module.exports = ({ nextOfKin, otherContacts }) =>
-  [...nextOfKin, ...otherContacts]
-    .filter(
-      ({ activeFlag, approvedVisitorFlag }) =>
-        activeFlag && approvedVisitorFlag,
-    )
-    .sort(sortContacts)
+module.exports = contacts =>
+  contacts
+    .filter(contact => contact.approvedVisitor)
+    .sort(sortBy('firstName'))
+    .sort(sortBy('lastName'))
```

Comment:
I think you need to filter by the approved flag?

Reply:
This does. line #5 ```contact.approvedVisitor```. The object returned contains a _approvedVisitor_ field as below.

```
{
    personId: 000000,
    firstName: 'firstname',
    lastName: 'lastname',
    dateOfBirth: '1998-09-22',
    relationshipCode: 'FRI',
    relationshipDescription: 'Friend',
    contactType: 'S',
    contactTypeDescription: 'Social/ Family',
    approvedVisitor: true,
    emergencyContact: false,
    nextOfKin: false,
    restrictions: [],
    addresses: [ [Object] ]
  },
```

Reply:
@BarryGeeMOJ sorry! Missed that 🤦

Reply:
no problem :)